scaleway/resource_bucket.go

```diff
@@ -0,0 +1,61 @@
+package scaleway
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+
+	api "github.com/nicolai86/scaleway-sdk"
+)
+
+func resourceScalewayBucket() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayBucketCreate,
+		Read:   resourceScalewayBucketRead,
+		Delete: resourceScalewayBucketDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the bucket",
+			},
+		},
+	}
+}
+
+func resourceScalewayBucketRead(d *schema.ResourceData, m interface{}) error {
+	scaleway := m.(*Client).scaleway
+
+	_, err := scaleway.ListObjects(d.Get("name").(string))
```

Comment:
so if this returns a not found error (e.g. a 404) we should clear the ID so this is detected and can be recreated e.g.

```
if err != nil {
  if resp.StatusCode == 404 {
    log.Printf("[DEBUG] Bucket %q was not found - removing from state!", name)
    d.SetId("")
    return nil
  }
}
```